EPRN.UnitTests/Portal/Controllers/WasteControllerTests.cs

```diff
@@ -445,5 +446,90 @@ public async Task SaveTonnes_WithInvalidData_ReturnsView()
             Assert.AreEqual(exportTonnageViewModel, viewResult.ViewData.Model);
             Assert.IsNull(null, viewResult.ViewName); // view is being returned as the same as the action
         }
+
+        [TestMethod]
+        public async Task BaledWithWire_ReturnCurrentView_WhenModelIsInvalid()
+        {
+            var baledWithWireModel = new BaledWithWireModel();
+            _mockWasteService.Setup(s => s.GetBaledWithWireModel(It.IsAny<int>())).ReturnsAsync(new BaledWithWireModel());
+
+            // Act
+            var result = await _wasteController.BaledWithWire(0);
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.IsInstanceOfType(result, typeof(ViewResult));
+
+            var viewResult = result as ViewResult;
+            Assert.IsNotNull(viewResult.ViewData.Model);
+
+            // check model is expected type
+            Assert.IsInstanceOfType(viewResult.ViewData.Model, typeof(BaledWithWireModel));
+
+            // check view name
+            Assert.IsNull(viewResult.ViewName);
+        }
+
+        [TestMethod]
+        public async Task BaledWithWire_Return_Correctly()
+        {
+            // Arrange
+            var baledWithWireModel = new BaledWithWireModel
+            {
+                JourneyId = 1,
+                BaledWithWire = true
+            };
+
+            _mockWasteService.Setup(s => s.GetBaledWithWireModel(1)).ReturnsAsync(baledWithWireModel);
+
+            // Act
+            var result = await _wasteController.BaledWithWire(1);
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.IsInstanceOfType(result, typeof(ViewResult));
+
+            var viewResult = result as ViewResult;
+            Assert.IsNotNull(viewResult.ViewData.Model);
+
+            // check model is expected type
+            Assert.IsInstanceOfType(viewResult.ViewData.Model, typeof(BaledWithWireModel));
+
+            // check view name
+            Assert.IsNull(viewResult.ViewName); // It's going to return the view name of the action by default
```

Comment:
The comment says it returns view name of the action by default but the code is asserting if viewResult.ViewName IsNull.

Reply:
This is the same as all the others.